File: CS/Form1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using System.Configuration;
using DevExpress.Xpo;
using DevExpress.Xpo.DB;
using DevExpress.Data.Filtering;

namespace AggregateFilterOnAssociatedCollection
{
	/// <summary>
	/// Summary description for Form1.
	/// </summary>
    public class Form1 : System.Windows.Forms.Form {
        private DevExpress.XtraEditors.PanelControl panelControl1;
        private DevExpress.XtraEditors.LabelControl labelControl1;
        private DevExpress.Xpo.UnitOfWork unitOfWork1;
        private DevExpress.Xpo.XPCollection xpCollection1;
        private DevExpress.XtraEditors.LabelControl labelControl2;
        private DevExpress.XtraGrid.GridControl gridControl1;
        private DevExpress.XtraGrid.Views.Grid.GridView gridView1;
        private DevExpress.XtraGrid.Columns.GridColumn colCustomerID;
        private DevExpress.XtraGrid.Columns.GridColumn colCompanyName;
        private DevExpress.XtraGrid.Columns.GridColumn colContactTitle;
        private DevExpress.XtraGrid.Views.Grid.GridView gridView2;
        private DevExpress.XtraGrid.Columns.GridColumn colOrderID;
        private DevExpress.XtraGrid.Columns.GridColumn colEmployeeFirstName;
        private DevExpress.XtraGrid.Columns.GridColumn colOrderDate;
        private DevExpress.XtraGrid.Columns.GridColumn colEmployeeLastName;
        private DevExpress.XtraGrid.Columns.GridColumn colFreight;
        private DevExpress.XtraEditors.SimpleButton btnFiterSimplified;
        private DevExpress.XtraEditors.SimpleButton btnClear;
        private DevExpress.XtraEditors.SimpleButton btnFilterStandard;
        /// <summary>
        /// Required designer variable.
        /// </summary>
        private System.ComponentModel.Container components = null;

        public Form1() {
            InitializeComponent();

        }

        /// <summary>
        /// Clean up any resources being used.
        /// </summary>
        protected override void Dispose( bool disposing ) {
            if( disposing ) {
                if (components != null) {
                    components.Dispose();
                }
            }
            base.Dispose( disposing );
        }

        #region Windows Form Designer generated code
        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent() {
            DevExpress.XtraGrid.GridLevelNode gridLevelNode1 = new DevExpress.XtraGrid.GridLevelNode();
            this.gridView2 = new DevExpress.XtraGrid.Views.Grid.GridView();
            this.colOrderID = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colEmployeeFirstName = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colEmployeeLastName = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colOrderDate = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colFreight = new DevExpress.XtraGrid.Columns.GridColumn();
            this.gridControl1 = new DevExpress.XtraGrid.GridControl();
            this.xpCollection1 = new DevExpress.Xpo.XPCollection();
            this.unitOfWork1 = new DevExpress.Xpo.UnitOfWork();
            this.gridView1 = new DevExpress.XtraGrid.Views.Grid.GridView();
            this.colCustomerID = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colCompanyName = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colContactTitle = new DevExpress.XtraGrid.Columns.GridColumn();
            this.panelControl1 = new DevExpress.XtraEditors.PanelControl();
            this.labelControl2 = new DevExpress.XtraEditors.LabelControl();
            this.labelControl1 = new DevExpress.XtraEditors.LabelControl();
            this.btnFilterStandard = new DevExpress.XtraEditors.SimpleButton();
            this.btnFiterSimplified = new DevExpress.XtraEditors.SimpleButton();
            this.btnClear = new DevExpress.XtraEditors.SimpleButton();
            ((System.ComponentModel.ISupportInitialize)(this.gridView2)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.gridControl1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.xpCollection1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.unitOfWork1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.gridView1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.panelControl1)).BeginInit();
            this.panelControl1.SuspendLayout();
            this.SuspendLayout();
            // 
            // gridView2
            // 
            this.gridView2.Columns.AddRange(new DevExpress.XtraGrid.Columns.GridColumn[] {
            this.colOrderID,
            this.colEmployeeFirstName,
            this.colEmployeeLastName,
            this.colOrderDate,
            this.colFreight});
            this.gridView2.GridControl = this.gridControl1;
            this.gridView2.Name = "gridView2";
            this.gridView2.OptionsView.ShowGroupPanel = false;
            // 
            // colOrderID
            // 
            this.colOrderID.Caption = "OrderID";
            this.colOrderID.FieldName = "OrderID";
            this.colOrderID.Name = "colOrderID";
            this.colOrderID.Visible = true;
            this.colOrderID.VisibleIndex = 0;
            // 
            // colEmployeeFirstName
            // 
            this.colEmployeeFirstName.Caption = "Employee First Name";
            this.colEmployeeFirstName.FieldName = "Employee.FirstName";
            this.colEmployeeFirstName.Name = "colEmployeeFirstName";
            this.colEmployeeFirstName.Visible = true;
            this.colEmployeeFirstName.VisibleIndex = 1;
            // 
            // colEmployeeLastName
            // 
            this.colEmployeeLastName.Caption = "Employee Last Name";
            this.colEmployeeLastName.FieldName = "Employee.LastName";
            this.colEmployeeLastName.Name = "colEmployeeLastName";
            this.colEmployeeLastName.Visible = true;
            this.colEmployeeLastName.VisibleIndex = 2;
            // 
            // colOrderDate
            // 
            this.colOrderDate.Caption = "Order Date";
            this.colOrderDate.FieldName = "OrderDate";
            this.colOrderDate.Name = "colOrderDate";
            this.colOrderDate.Visible = true;
            this.colOrderDate.VisibleIndex = 3;
            // 
            // colFreight
            // 
            this.colFreight.Caption = "Freight";
            this.colFreight.FieldName = "Freight";
            this.colFreight.Name = "colFreight";
            this.colFreight.Visible = true;
            this.colFreight.VisibleIndex = 4;
            // 
            // gridControl1
            // 
            this.gridControl1.DataSource = this.xpCollection1;
            this.gridControl1.Dock = System.Windows.Forms.DockStyle.Fill;
            gridLevelNode1.LevelTemplate = this.gridView2;
            gridLevelNode1.RelationName = "Orders";
            this.gridControl1.LevelTree.Nodes.AddRange(new DevExpress.XtraGrid.GridLevelNode[] {
            gridLevelNode1});
            this.gridControl1.Location = new System.Drawing.Point(0, 0);
            this.gridControl1.MainView = this.gridView1;
            this.gridControl1.Name = "gridControl1";
            this.gridControl1.ShowOnlyPredefinedDetails = true;
            this.gridControl1.Size = new System.Drawing.Size(547, 279);
            this.gridControl1.TabIndex = 1;
            this.gridControl1.ViewCollection.AddRange(new DevExpress.XtraGrid.Views.Base.BaseView[] {
            this.gridView1,
            this.gridView2});
            // 
            // xpCollection1
            // 
            this.xpCollection1.DisplayableProperties = "This;CustomerID;CompanyName;ContactTitle;Orders";
            this.xpCollection1.ObjectType = typeof(AggregateFilterOnAssociatedCollection.Customer);
            this.xpCollection1.Session = this.unitOfWork1;
            // 
            // gridView1
            // 
            this.gridView1.Columns.AddRange(new DevExpress.XtraGrid.Columns.GridColumn[] {
            this.colCustomerID,
            this.colCompanyName,
            this.colContactTitle});
            this.gridView1.GridControl = this.gridControl1;
            this.gridView1.Name = "gridView1";
            this.gridView1.OptionsDetail.ShowDetailTabs = false;
            this.gridView1.OptionsView.ShowFooter = true;
            this.gridView1.OptionsView.ShowGroupPanel = false;
            // 
            // colCustomerID
            // 
            this.colCustomerID.Caption = "CustomerID";
            this.colCustomerID.FieldName = "CustomerID";
            this.colCustomerID.Name = "colCustomerID";
            this.colCustomerID.SummaryItem.SummaryType = DevExpress.Data.SummaryItemType.Count;
            this.colCustomerID.Visible = true;
            this.colCustomerID.VisibleIndex = 0;
            // 
            // colCompanyName
            // 
            this.colCompanyName.Caption = "CompanyName";
            this.colCompanyName.FieldName = "CompanyName";
            this.colCompanyName.Name = "colCompanyName";
            this.colCompanyName.Visible = true;
            this.colCompanyName.VisibleIndex = 1;
            // 
            // colContactTitle
            // 
            this.colContactTitle.Caption = "ContactTitle";
            this.colContactTitle.FieldName = "ContactTitle";
            this.colContactTitle.Name = "colContactTitle";
            this.colContactTitle.Visible = true;
            this.colContactTitle.VisibleIndex = 2;
            // 
            // panelControl1
            // 
            this.panelControl1.Controls.Add(this.labelControl2);
            this.panelControl1.Controls.Add(this.labelControl1);
            this.panelControl1.Controls.Add(this.btnFilterStandard);
            this.panelControl1.Controls.Add(this.btnFiterSimplified);
            this.panelControl1.Controls.Add(this.btnClear);
            this.panelControl1.Dock = System.Windows.Forms.DockStyle.Bottom;
            this.panelControl1.Location = new System.Drawing.Point(0, 279);
            this.panelControl1.Name = "panelControl1";
            this.panelControl1.Size = new System.Drawing.Size(547, 88);
            this.panelControl1.TabIndex = 0;
            // 
            // labelControl2
            // 
            this.labelControl2.AutoSizeMode = DevExpress.XtraEditors.LabelAutoSizeMode.Horizontal;
            this.labelControl2.Location = new System.Drawing.Point(16, 64);
            this.labelControl2.Name = "labelControl2";
            this.labelControl2.Size = new System.Drawing.Size(63, 13);
            this.labelControl2.TabIndex = 4;
            this.labelControl2.Text = "labelControl2";
            // 
            // labelControl1
            // 
            this.labelControl1.AutoSizeMode = DevExpress.XtraEditors.LabelAutoSizeMode.Horizontal;
            this.labelControl1.Location = new System.Drawing.Point(16, 40);
            this.labelControl1.Name = "labelControl1";
            this.labelControl1.Size = new System.Drawing.Size(63, 13);
            this.labelControl1.TabIndex = 3;
            this.labelControl1.Text = "labelControl1";
            // 
            // btnFilterStandard
            // 
            this.btnFilterStandard.Location = new System.Drawing.Point(18, 8);
            this.btnFilterStandard.Name = "btnFilterStandard";
            this.btnFilterStandard.Size = new System.Drawing.Size(184, 23);
            this.btnFilterStandard.TabIndex = 2;
            this.btnFilterStandard.Text = "Apply Filter (Standard Syntax)";
            this.btnFilterStandard.Click += new System.EventHandler(this.btnFiterStandard_Click);
            // 
            // btnFiterSimplified
            // 
            this.btnFiterSimplified.Location = new System.Drawing.Point(208, 8);
            this.btnFiterSimplified.Name = "btnFiterSimplified";
            this.btnFiterSimplified.Size = new System.Drawing.Size(184, 23);
            this.btnFiterSimplified.TabIndex = 2;
            this.btnFiterSimplified.Text = "Apply Filter (Simplified Syntax)";
            this.btnFiterSimplified.Click += new System.EventHandler(this.btnFiterSimplified_Click);
            // 
            // btnClear
            // 
            this.btnClear.Location = new System.Drawing.Point(400, 8);
            this.btnClear.Name = "btnClear";
            this.btnClear.Size = new System.Drawing.Size(75, 23);
            this.btnClear.TabIndex = 1;
            this.btnClear.Text = "Clear Filter";
            this.btnClear.Click += new System.EventHandler(this.btnClear_Click);
            // 
            // Form1
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.ClientSize = new System.Drawing.Size(547, 367);
            this.Controls.Add(this.gridControl1);
            this.Controls.Add(this.panelControl1);
            this.Name = "Form1";
            this.Text = "Form1";
            this.Load += new System.EventHandler(this.Form1_Load);
            ((System.ComponentModel.ISupportInitialize)(this.gridView2)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.gridControl1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.xpCollection1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.unitOfWork1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.gridView1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.panelControl1)).EndInit();
            this.panelControl1.ResumeLayout(false);
            this.panelControl1.PerformLayout();
            this.ResumeLayout(false);

        }
        #endregion

        /// <summary>
        /// The main entry point for the application.
        /// </summary>
        [STAThread]
        static void Main() {
            string conn = MSSqlConnectionProvider.GetConnectionString("(local)", "Northwind");
            XpoDefault.DataLayer = XpoDefault.GetDataLayer(conn, AutoCreateOption.SchemaAlreadyExists);

            Application.Run(new Form1());
        }
        
        private void Form1_Load(object sender, System.EventArgs e) {
            SetFilter(null);
        }

        void SetFilter(CriteriaOperator criteria) {
            xpCollection1.Criteria = criteria;

            if(ReferenceEquals(criteria, null))
                labelControl1.Text = string.Empty;
            else
                labelControl1.Text = criteria.ToString();                
            labelControl2.Text = string.Format("Total row count: {0}", xpCollection1.Count);
        }

        private void btnFiterStandard_Click(object sender, System.EventArgs e) {
            CriteriaOperator criteria = new AggregateOperand("Orders", Aggregate.Count) > 0;
            SetFilter(criteria);
        }

        private void btnFiterSimplified_Click(object sender, System.EventArgs e) {
            CriteriaOperator criteria = new OperandProperty("Orders")[null].Count() > 0;
            SetFilter(criteria);
        }

        

        private void btnClear_Click(object sender, System.EventArgs e) {
            SetFilter(null);
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Data
Imports System.Configuration
Imports DevExpress.Xpo
Imports DevExpress.Xpo.DB
Imports DevExpress.Data.Filtering

''' <summary>
''' Summary description for Form1.
''' </summary>
Public Class Form1
    Inherits System.Windows.Forms.Form
    Private panelControl1 As DevExpress.XtraEditors.PanelControl
    Private labelControl1 As DevExpress.XtraEditors.LabelControl
    Private unitOfWork1 As DevExpress.Xpo.UnitOfWork
    Private xpCollection1 As DevExpress.Xpo.XPCollection
    Private labelControl2 As DevExpress.XtraEditors.LabelControl
    Private gridControl1 As DevExpress.XtraGrid.GridControl
    Private gridView1 As DevExpress.XtraGrid.Views.Grid.GridView
    Private colCustomerID As DevExpress.XtraGrid.Columns.GridColumn
    Private colCompanyName As DevExpress.XtraGrid.Columns.GridColumn
    Private colContactTitle As DevExpress.XtraGrid.Columns.GridColumn
    Private gridView2 As DevExpress.XtraGrid.Views.Grid.GridView
    Private colOrderID As DevExpress.XtraGrid.Columns.GridColumn
    Private colEmployeeFirstName As DevExpress.XtraGrid.Columns.GridColumn
    Private colOrderDate As DevExpress.XtraGrid.Columns.GridColumn
    Private colEmployeeLastName As DevExpress.XtraGrid.Columns.GridColumn
    Private colFreight As DevExpress.XtraGrid.Columns.GridColumn
    Private WithEvents btnFiterSimplified As DevExpress.XtraEditors.SimpleButton
    Private WithEvents btnClear As DevExpress.XtraEditors.SimpleButton
    Private WithEvents btnFilterStandard As DevExpress.XtraEditors.SimpleButton
    ''' <summary>
    ''' Required designer variable.
    ''' </summary>
    Private components As System.ComponentModel.Container = Nothing

    Public Sub New()
        InitializeComponent()

    End Sub

    ''' <summary>
    ''' Clean up any resources being used.
    ''' </summary>
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not components Is Nothing Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub

#Region "Windows Form Designer generated code"
    ''' <summary>
    ''' Required method for Designer support - do not modify
    ''' the contents of this method with the code editor.
    ''' </summary>
    Private Sub InitializeComponent()
        Dim GridLevelNode1 As DevExpress.XtraGrid.GridLevelNode = New DevExpress.XtraGrid.GridLevelNode
        Me.gridView2 = New DevExpress.XtraGrid.Views.Grid.GridView
        Me.colOrderID = New DevExpress.XtraGrid.Columns.GridColumn
        Me.colEmployeeFirstName = New DevExpress.XtraGrid.Columns.GridColumn
        Me.colEmployeeLastName = New DevExpress.XtraGrid.Columns.GridColumn
        Me.colOrderDate = New DevExpress.XtraGrid.Columns.GridColumn
        Me.colFreight = New DevExpress.XtraGrid.Columns.GridColumn
        Me.gridControl1 = New DevExpress.XtraGrid.GridControl
        Me.xpCollection1 = New DevExpress.Xpo.XPCollection
        Me.unitOfWork1 = New DevExpress.Xpo.UnitOfWork
        Me.gridView1 = New DevExpress.XtraGrid.Views.Grid.GridView
        Me.colCustomerID = New DevExpress.XtraGrid.Columns.GridColumn
        Me.colCompanyName = New DevExpress.XtraGrid.Columns.GridColumn
        Me.colContactTitle = New DevExpress.XtraGrid.Columns.GridColumn
        Me.panelControl1 = New DevExpress.XtraEditors.PanelControl
        Me.labelControl2 = New DevExpress.XtraEditors.LabelControl
        Me.labelControl1 = New DevExpress.XtraEditors.LabelControl
        Me.btnFilterStandard = New DevExpress.XtraEditors.SimpleButton
        Me.btnFiterSimplified = New DevExpress.XtraEditors.SimpleButton
        Me.btnClear = New DevExpress.XtraEditors.SimpleButton
        CType(Me.gridView2, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.gridControl1, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.xpCollection1, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.unitOfWork1, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.gridView1, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.panelControl1, System.ComponentModel.ISupportInitialize).BeginInit()
        Me.panelControl1.SuspendLayout()
        Me.SuspendLayout()
        '
        'gridView2
        '
        Me.gridView2.Columns.AddRange(New DevExpress.XtraGrid.Columns.GridColumn() {Me.colOrderID, Me.colEmployeeFirstName, Me.colEmployeeLastName, Me.colOrderDate, Me.colFreight})
        Me.gridView2.GridControl = Me.gridControl1
        Me.gridView2.Name = "gridView2"
        Me.gridView2.OptionsView.ShowGroupPanel = False
        '
        'colOrderID
        '
        Me.colOrderID.Caption = "OrderID"
        Me.colOrderID.FieldName = "OrderID"
        Me.colOrderID.Name = "colOrderID"
        Me.colOrderID.Visible = True
        Me.colOrderID.VisibleIndex = 0
        '
        'colEmployeeFirstName
        '
        Me.colEmployeeFirstName.Caption = "Employee First Name"
        Me.colEmployeeFirstName.FieldName = "Employee.FirstName"
        Me.colEmployeeFirstName.Name = "colEmployeeFirstName"
        Me.colEmployeeFirstName.Visible = True
        Me.colEmployeeFirstName.VisibleIndex = 1
        '
        'colEmployeeLastName
        '
        Me.colEmployeeLastName.Caption = "Employee Last Name"
        Me.colEmployeeLastName.FieldName = "Employee.LastName"
        Me.colEmployeeLastName.Name = "colEmployeeLastName"
        Me.colEmployeeLastName.Visible = True
        Me.colEmployeeLastName.VisibleIndex = 2
        '
        'colOrderDate
        '
        Me.colOrderDate.Caption = "Order Date"
        Me.colOrderDate.FieldName = "OrderDate"
        Me.colOrderDate.Name = "colOrderDate"
        Me.colOrderDate.Visible = True
        Me.colOrderDate.VisibleIndex = 3
        '
        'colFreight
        '
        Me.colFreight.Caption = "Freight"
        Me.colFreight.FieldName = "Freight"
        Me.colFreight.Name = "colFreight"
        Me.colFreight.Visible = True
        Me.colFreight.VisibleIndex = 4
        '
        'gridControl1
        '
        Me.gridControl1.DataSource = Me.xpCollection1
        Me.gridControl1.Dock = System.Windows.Forms.DockStyle.Fill
        GridLevelNode1.LevelTemplate = Me.gridView2
        GridLevelNode1.RelationName = "Orders"
        Me.gridControl1.LevelTree.Nodes.AddRange(New DevExpress.XtraGrid.GridLevelNode() {GridLevelNode1})
        Me.gridControl1.Location = New System.Drawing.Point(0, 0)
        Me.gridControl1.MainView = Me.gridView1
        Me.gridControl1.Name = "gridControl1"
        Me.gridControl1.ShowOnlyPredefinedDetails = True
        Me.gridControl1.Size = New System.Drawing.Size(547, 279)
        Me.gridControl1.TabIndex = 1
        Me.gridControl1.ViewCollection.AddRange(New DevExpress.XtraGrid.Views.Base.BaseView() {Me.gridView1, Me.gridView2})
        '
        'xpCollection1
        '
        Me.xpCollection1.DisplayableProperties = "This;CustomerID;CompanyName;ContactTitle;Orders"
        Me.xpCollection1.ObjectType = GetType(AggregateFilterOnAssociatedCollection.Customer)
        Me.xpCollection1.Session = Me.unitOfWork1
        '
        'gridView1
        '
        Me.gridView1.Columns.AddRange(New DevExpress.XtraGrid.Columns.GridColumn() {Me.colCustomerID, Me.colCompanyName, Me.colContactTitle})
        Me.gridView1.GridControl = Me.gridControl1
        Me.gridView1.Name = "gridView1"
        Me.gridView1.OptionsDetail.ShowDetailTabs = False
        Me.gridView1.OptionsView.ShowFooter = True
        Me.gridView1.OptionsView.ShowGroupPanel = False
        '
        'colCustomerID
        '
        Me.colCustomerID.Caption = "CustomerID"
        Me.colCustomerID.FieldName = "CustomerID"
        Me.colCustomerID.Name = "colCustomerID"
        Me.colCustomerID.SummaryItem.SummaryType = DevExpress.Data.SummaryItemType.Count
        Me.colCustomerID.Visible = True
        Me.colCustomerID.VisibleIndex = 0
        '
        'colCompanyName
        '
        Me.colCompanyName.Caption = "CompanyName"
        Me.colCompanyName.FieldName = "CompanyName"
        Me.colCompanyName.Name = "colCompanyName"
        Me.colCompanyName.Visible = True
        Me.colCompanyName.VisibleIndex = 1
        '
        'colContactTitle
        '
        Me.colContactTitle.Caption = "ContactTitle"
        Me.colContactTitle.FieldName = "ContactTitle"
        Me.colContactTitle.Name = "colContactTitle"
        Me.colContactTitle.Visible = True
        Me.colContactTitle.VisibleIndex = 2
        '
        'panelControl1
        '
        Me.panelControl1.Controls.Add(Me.labelControl2)
        Me.panelControl1.Controls.Add(Me.labelControl1)
        Me.panelControl1.Controls.Add(Me.btnFilterStandard)
        Me.panelControl1.Controls.Add(Me.btnFiterSimplified)
        Me.panelControl1.Controls.Add(Me.btnClear)
        Me.panelControl1.Dock = System.Windows.Forms.DockStyle.Bottom
        Me.panelControl1.Location = New System.Drawing.Point(0, 279)
        Me.panelControl1.Name = "panelControl1"
        Me.panelControl1.Size = New System.Drawing.Size(547, 88)
        Me.panelControl1.TabIndex = 0
        '
        'labelControl2
        '
        Me.labelControl2.AutoSizeMode = DevExpress.XtraEditors.LabelAutoSizeMode.Horizontal
        Me.labelControl2.Location = New System.Drawing.Point(16, 64)
        Me.labelControl2.Name = "labelControl2"
        Me.labelControl2.Size = New System.Drawing.Size(63, 13)
        Me.labelControl2.TabIndex = 4
        Me.labelControl2.Text = "labelControl2"
        '
        'labelControl1
        '
        Me.labelControl1.AutoSizeMode = DevExpress.XtraEditors.LabelAutoSizeMode.Horizontal
        Me.labelControl1.Location = New System.Drawing.Point(16, 40)
        Me.labelControl1.Name = "labelControl1"
        Me.labelControl1.Size = New System.Drawing.Size(63, 13)
        Me.labelControl1.TabIndex = 3
        Me.labelControl1.Text = "labelControl1"
        '
        'btnFilterStandard
        '
        Me.btnFilterStandard.Location = New System.Drawing.Point(18, 8)
        Me.btnFilterStandard.Name = "btnFilterStandard"
        Me.btnFilterStandard.Size = New System.Drawing.Size(184, 23)
        Me.btnFilterStandard.TabIndex = 2
        Me.btnFilterStandard.Text = "Apply Filter (Standard Syntax)"
        '
        'btnFiterSimplified
        '
        Me.btnFiterSimplified.Location = New System.Drawing.Point(208, 8)
        Me.btnFiterSimplified.Name = "btnFiterSimplified"
        Me.btnFiterSimplified.Size = New System.Drawing.Size(184, 23)
        Me.btnFiterSimplified.TabIndex = 2
        Me.btnFiterSimplified.Text = "Apply Filter (Simplified Syntax)"
        '
        'btnClear
        '
        Me.btnClear.Location = New System.Drawing.Point(400, 8)
        Me.btnClear.Name = "btnClear"
        Me.btnClear.Size = New System.Drawing.Size(75, 23)
        Me.btnClear.TabIndex = 1
        Me.btnClear.Text = "Clear Filter"
        '
        'Form1
        '
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.ClientSize = New System.Drawing.Size(547, 367)
        Me.Controls.Add(Me.gridControl1)
        Me.Controls.Add(Me.panelControl1)
        Me.Name = "Form1"
        Me.Text = "Form1"
        CType(Me.gridView2, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.gridControl1, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.xpCollection1, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.unitOfWork1, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.gridView1, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.panelControl1, System.ComponentModel.ISupportInitialize).EndInit()
        Me.panelControl1.ResumeLayout(False)
        Me.panelControl1.PerformLayout()
        Me.ResumeLayout(False)

    End Sub
#End Region

    ''' <summary>
    ''' The main entry point for the application.
    ''' </summary>
    <STAThread()> _
    Shared Sub Main()
        Dim conn As String = MSSqlConnectionProvider.GetConnectionString("(local)", "Northwind")
        XpoDefault.DataLayer = XpoDefault.GetDataLayer(conn, AutoCreateOption.SchemaAlreadyExists)

        Application.Run(New Form1())
    End Sub

    Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
        SetFilter(Nothing)
    End Sub

    Private Sub SetFilter(ByVal criteria As CriteriaOperator)
        xpCollection1.Criteria = criteria

        If ReferenceEquals(criteria, Nothing) Then
            labelControl1.Text = String.Empty
        Else
            labelControl1.Text = criteria.ToString()
        End If
        labelControl2.Text = String.Format("Total row count: {0}", xpCollection1.Count)
    End Sub

    Private Sub btnFiterStandard_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles btnFilterStandard.Click
        Dim criteria As CriteriaOperator = New AggregateOperand("Orders", Aggregate.Count) > 0
        SetFilter(criteria)
    End Sub

    Private Sub btnFiterSimplified_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles btnFiterSimplified.Click
        Dim criteria As CriteriaOperator = New OperandProperty("Orders").Item(Nothing).Count() > 0
        SetFilter(criteria)
    End Sub

    Private Sub btnClear_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles btnClear.Click
        SetFilter(Nothing)
    End Sub
End Class
